coffee_maker_navigator_2/lib/di/di/dependency_containers/dependency_container.dart

```diff
@@ -0,0 +1,55 @@
+/// An abstract class representing a dependency container.
+///
+/// The `DependencyContainer` class serves as a base class for all containers
+/// that manage dependencies in the Dependency Injection (DI) system. It defines
+/// a common interface that all concrete dependency containers must implement.
+///
+/// The primary responsibility of a dependency container is to manage the lifecycle of the
+/// dependencies it holds. This includes initializing and disposing of dependencies as needed.
+///
+/// The `dispose` method is a crucial part of this class, as it ensures that
+/// resources held by the container are properly released when the container
+/// is no longer needed.
+abstract class DependencyContainer {
+  /// Disposes of the dependencies managed by this container.
+  ///
+  /// This method should be implemented by concrete dependency containers to
+  /// release any resources they hold. It is called when the container is no
+  /// longer needed, ensuring that resources are properly cleaned up.
+  void dispose();
+}
+
+/// An abstract class representing a dependency container that supports
+/// asynchronous initialization.
+///
+/// The `AsyncDependencyContainer` class extends `DependencyContainer` and adds
+/// an `init` method that returns a `Future<void>`. This allows for asynchronous
+/// initialization of dependencies.
+abstract class AsyncDependencyContainer extends DependencyContainer {
```

Comment:
It is can be only for app level, others might not have a time between subscribing and get container.
But I like the idea of having an interface for containers. Maybe we can describe which dependencies can be requested from the concrete container.

Reply:
This depends on where we execute the `onInit` call. Some dependencies would require async initialization of its dependencies before they could be initialized.